src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,121 @@
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            LOGGER.info("Product with id=" + product.getId() + " was created");
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Creation of product is failed", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * from internet_shop.products where product_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getCopyOfProduct(resultSet));
+            }
+            LOGGER.info("Product with id=" + id + " was received");
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of product with id="
+                    + id + " is failed", e);
+
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        List<Product> products = new ArrayList<>();
+        String query = "SELECT * FROM internet_shop.products;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                products.add(getCopyOfProduct(resultSet));
+            }
+            LOGGER.info("List of products was received");
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting of list of products is failed", e);
+        }
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE internet_shop.products SET name=?, price=? WHERE product_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setBigDecimal(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            statement.executeUpdate();
+            LOGGER.info("Product with id=" + product.getId() + " was updated");
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Update of product with id="
+                    + product.getId() + " is failed", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM internet_shop.products WHERE product_id=?;";
+        try (Connection connection = ConnectionUtil.getConnection();) {
```

Comment:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection()) {
```
I guess the semicolon is redundant

Reply:
I guess  so too)